rest_framework/fields.py

```diff
@@ -1027,6 +1028,11 @@ def __init__(self, max_digits, decimal_places, coerce_to_string=None, max_value=
             self.validators.append(
                 MinValueValidator(self.min_value, message=message))
 
+        valid_roundings = [v for k, v in vars(decimal).items() if k.startswith('ROUND_')]
+        if rounding is not None and rounding not in valid_roundings:
+            self.fail('invalid_rounding', rounding=rounding, valid_roundings=valid_roundings)
+        self.rounding = rounding
+
```

Comment:
It looks like you're trying to validate that the user has passed a valid `rounding` setting to the field. In this case, you want to use a standard `assert` statement (for example, see the `required`/`default` checks [here](https://github.com/encode/django-rest-framework/blob/3.7.1/rest_framework/fields.py#L319-L322)). The `self.fail()` method is intended to raise validation errors when processing the value during deserialization.

Reply:
Hi @rpkilby, thanks for reviewing. I have updated accordingly. Hope it works out.

Reply:
Is validation necessary?  If someone passes a bad parameter, things will blow up as soon as they try out the serializer.

Reply:
If they forget to try out the serializer, checking upfront notifies the problem as soon as the server is started.